core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -837,16 +837,19 @@ public final void checkTotalOrderSupported() {
       Address excludedTarget = deliverOrder == DeliverOrder.TOTAL ? null : getAddress();
       MultiTargetRequest<T> request =
             new MultiTargetRequest<>(collector, requestId, requests, targets, excludedTarget);
-      try {
-         addRequest(request);
-         boolean checkView = request.onNewView(clusterView.getMembersSet());
-         sendCommand(targets, command, requestId, deliverOrder, isRsvpCommand(command), checkView);
-      } catch (Throwable t) {
-         request.cancel(true);
-         throw t;
-      }
-      if (timeout > 0) {
-         request.setTimeout(timeoutExecutor, timeout, unit);
+      // Request may be completed due to exclusion of target nodes, so only send it if it isn't complete
+      if (!request.isDone()) {
```

Comment:
I would suggest `if (request.isDone() return request;` to avoid indenting the rest of the method.

Reply:
Sure.